packages/oidc-middleware/src/ExpressOIDC.js

```diff
@@ -68,28 +79,51 @@ module.exports = class ExpressOIDC extends EventEmitter {
     // Validate the client_secret param
     assertClientSecret(client_secret);
 
-    // Validate the redirect_uri param
-    assertRedirectUri(redirect_uri);
+    
+    // DEPRECATED: options.routes.callback is deprecated in favor of options.routes.loginCallback
+    // Fallbacks for deprecated options will be removed in a future update
+    if(options.routes) { 
+      options.routes.loginCallback = options.routes.loginCallback || options.routes.callback;
+    }
 
     // Add defaults to the options
-    options = _.merge({
+    options = merge({
       response_type: 'code',
       scope: 'openid',
       routes: {
         login: {
           path: '/login'
         },
-        callback: {
+        loginCallback: {
           path: '/authorization-code/callback',
-          defaultRedirect: '/'
-        }
+          afterCallback: '/'
+        },
+        logout: { 
+          path: '/logout'
+        },
+        logoutCallback: { 
+          path: '/logout/callback',
+          afterCallback: '/'
+        },
       },
-      sessionKey: sessionKey || `oidc:${options.issuer}`,
+      sessionKey: sessionKey || `oidc:${issuer}`,
       maxClockSkew: 120
-    }, options)
+    }, options);
+
+    // Build redirect uris unless they were explicitly set
+    // DEPRECATED: `redirect_uri` is deprecated in favor of `loginRedirectUri` (and most users will just set `appBaseUrl` and not set either of these)
+    // DEPRECATED: `options.routes.callback` is deprecated in favor of `options.routes.loginCallback`
+    // Fallbacks for deprecated options will be removed in a future update
+    options.loginRedirectUri = loginRedirectUri || redirect_uri || `${appBaseUrl}${options.routes.loginCallback.path}` || `${appBaseUrl}${options.routes.callback.path}`;
```

Comment:
nit: maybe worth to put option.routes in a const too

Reply:
Good point.